### PR TITLE
Bump search native; Search 2.6.1 release

### PR DESCRIPTION
**Pull request by DzmitryFomchyn**
### Description
- Bump Search native to 2.6.2, Search SDK release
- Introduce Details Api

### Checklist
- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have updated the `CHANGELOG` including this PR (where applicable)
- [ ] I have run tests and automatic checks locally
- [ ] I have run `pitest` check locally and checked that the coverage increased (or didn't change) or decreased insignificantly
- [ ] I have made corresponding changes to the documentation (where applicable)
- [ ] I have grouped commits logically or I promise to squash my commits before merge

### CODE REVIEW

File: MapboxSearch/sdk/src/main/java/com/mapbox/search/details/DetailsApi.kt [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+    public companion object {
+
+        /**
+         * Creates a new instance of the [DetailsApi].
+         *
+         * @param settings [DetailsApiSettings] settings.
+         * @return a new instance instance of the [DetailsApi].
+         */
+        @JvmStatic
+        public fun create(settings: DetailsApiSettings): DetailsApi {
```
Comment (mikeringrose) [inline MapboxSearch/sdk/src/main/java/com/mapbox/search/details/DetailsApi.kt:77]: nit: make settings optional
```
public fun create(
    settings: DetailsApiSettings = DetailsApiSettings()
)
...
```

File: MapboxSearch/sdk/src/main/java/com/mapbox/search/details/RetrieveDetailsOptions.kt [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+
+    /**
+     * @suppress
+     */
+    override fun hashCode(): Int {
+        var result = attributeSets?.hashCode() ?: 0
+        result = 31 * result + language.hashCode()
+        result = 31 * result + (worldview?.hashCode() ?: 0)
+        return result
+    }
```
Comment (mikeringrose) [inline MapboxSearch/sdk/src/main/java/com/mapbox/search/details/RetrieveDetailsOptions.kt:66]: nit: could just be
```
override fun hashCode(): Int = listOf(attributeSets.orEmpty(), language, worldview).hashCode()
```

**Review by mikeringrose — DISMISSED** [review]

File: MapboxSearch/sdk/src/main/java/com/mapbox/search/details/RetrieveDetailsOptions.kt (hunk 2 above)
Comment (DzmitryFomchyn) [reply MapboxSearch/sdk/src/main/java/com/mapbox/search/details/RetrieveDetailsOptions.kt:66]: It's autogenerated by the IDE, I'd left it as is.

**Review by mikeringrose — APPROVED** [review]

**Comment by DzmitryFomchyn** [comment]
Not all the integration tests are passing, it's a known issue, skipping this for now.

**Pull request merged**